app/src/main/java/com/example/evently/ui/common/ConfirmFragmentTextInput.java

```diff
@@ -119,6 +124,9 @@ private void connectUI(View view, Bundle args) {
         TextInputEditText textEdit = view.findViewById(R.id.text_field);
         Button cancel = view.findViewById(R.id.cancel_button);
         Button confirm = view.findViewById(R.id.confirm_button);
+        TextInputValidator validateType = (TextInputValidator) args.getSerializable(validateKey);
+
+        if (validateType == null) throw new RuntimeException("No given validation type");
```

Comment:
You can actually use `.valueOf()` since all enums have it. So instead of putting serializable you can `putString` with `.name()` on the enum.

The two functions are inverses of each other
```haskell
valueOf . name == name . valueOf == id
```